File: Domain/Entities/Expense.cs
using Domain.Common;

namespace Domain.Entities;

public class Expense : BaseEntity
{
    public Guid StoreId { get; private set; }

    public string Category { get; private set; } = string.Empty;

    public decimal Amount { get; private set; }

    public DateTime IncurredAt { get; private set; }

    public string? Note { get; private set; }

    private Expense()
    {
    }

    public Expense(
        Guid storeId,
        string category,
        decimal amount,
        DateTime incurredAt,
        string? note = null)
    {
        StoreId = storeId;
        Category = category;
        Amount = amount;
        IncurredAt = incurredAt;
        Note = note;
    }
}

File: Frontend/src/app/core/models/expense.ts
export interface Expense {
  id: string;
  storeId: string;
  category: string;
  amount: number;
  incurredAt: string;
  note: string | null;
}

export interface RecordExpenseRequest {
  category: string;
  amount: number;
  incurredAt: string;
  note?: string | null;
}